R/tufte_handout.R

```diff
@@ -56,26 +46,50 @@ tufte_handout <- function(fig_width = 4,
   
   # set hooks for special plot output
   knitr_options$knit_hooks$plot <- function(x, options) {
-    
-    # determine caption (if any)
-    caption <- ifelse(is.null(options$fig.cap), 
-                      "",
-                      paste("\\caption{", options$fig.cap, "}\n", sep = ""))
-    
+
     # determine figure type
     if (isTRUE(options$fig.margin)) 
-      figtype <- "marginfigure"
+      options$fig.env <- "marginfigure"
     else if (isTRUE(options$fig.fullwidth))
-      figtype <- "figure*"
-    else
-      figtype <- "figure"
-    
-    # return the latex
-    paste(sprintf('\\begin{%s}\n \\includegraphics{%s}\n%s\\end{%s}\n',
-                  figtype, x, caption, figtype))
+      options$fig.env <- "figure*"
+
+    knitr::hook_plot_tex(x, options)
   }
   
   # override the knitr settings of the base format and return the format
   format$knitr <- knitr_options
   format
 }
+
+#' @details \code{tufte_html()} provides the HTML format based on the Tufte CSS:
+#'   \url{https://edwardtufte.github.io/tufte-css/}.
+#' @param css One or more CSS files to include, and the Tufte CSS will always be
+#'   included in this list
+#' @rdname tufte_handout
+#' @export
+tufte_html <- function(css = NULL, ...) {
```

Comment:
We don't have a consistent style on warning but I agree that it's good form
to use call. = FALSE.

On Tue, Dec 29, 2015 at 9:12 AM, Hadley Wickham <notifications@github.com>
wrote:

> In R/tufte_handout.R
> <https://github.com/rstudio/rmarkdown/pull/581#discussion_r48542368>:
>
> >    }
> >
> >    # override the knitr settings of the base format and return the format
> >    format$knitr <- knitr_options
> >    format
> >  }
> > +
> > +#' @details \code{tufte_html()} provides the HTML format based on the Tufte CSS:
> > +#'   \url{https://edwardtufte.github.io/tufte-css/}.
> > +#' @param css One or more CSS files to include, and the Tufte CSS will always be
> > +#'   included in this list
> > +#' @rdname tufte_handout
> > +#' @export
> > +tufte_html <- function(css = NULL, ...) {
>
> I think using character() instead of NULL is slightly better as it
> suggests the type of vector you should be supplying.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rstudio/rmarkdown/pull/581/files#r48542368>.
>